PublicGitArchive/pga-create/cmd/pga-create/select.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/pkg/errors"
+	pb "gopkg.in/cheggaaa/pb.v1"
+)
+
+type selectCommand struct {
+	Stars           string `short:"s" long:"stars" required:"true" description:"Input path for the file with the numbers of stars per repository."`
+	Languages       string `short:"g" long:"languages" description:"Input path for the gzipped file with the mapping between languages and repositories."`
+	Repositories    string `short:"r" long:"repositories" required:"true" description:"Input path for the gzipped file with the repository names and identifiers."`
+	MinStars        int    `short:"m" long:"min-stars" description:"Minimum number of stars."`
+	Max             int    `short:"n" long:"max" default:"-1" description:"Maximum number of top-starred repositories to clone. -1 means unlimited. Language filter is applied before."`
+	FilterLanguages string `short:"l" long:"filter-languages" description:"Comma separated list of languages."`
+	UrlTemplate     string `long:"url-template" default:"git://github.com/%s.git" description:"Output URL printf template."`
+}
+
+func (c *selectCommand) Execute(args []string) error {
+	var filteredLangsSplitted []string
+	if len(c.FilterLanguages) == 0 {
+		filteredLangsSplitted = []string{}
+	} else {
+		filteredLangsSplitted = strings.Split(c.FilterLanguages, ",")
+	}
+
+	selectRepos(selectionParameters{
+		StarsFile:         c.Stars,
+		LanguagesFile:     c.Languages,
+		ReposFile:         c.Repositories,
+		MinStars:          c.MinStars,
+		FilteredLanguages: filteredLangsSplitted,
+		TopN:              c.Max,
+		URLTemplate:       c.UrlTemplate,
+	})
+
+	return nil
+}
+
+type selectionParameters struct {
+	StarsFile         string
+	LanguagesFile     string
+	ReposFile         string
+	FilteredLanguages []string
+	MinStars          int
+	TopN              int
+	URLTemplate       string
+}
+
+func selectRepos(parameters selectionParameters) {
```

Comment:
would it make more sense to have this return an error rather than call `fail` internally?